packages/knex/src/AbstractSqlDriver.ts

```diff
@@ -473,10 +475,18 @@ export abstract class AbstractSqlDriver<C extends AbstractSqlConnection = Abstra
       const field = parentTableAlias ? `${parentTableAlias}.${prop.name}` : prop.name;
       const path = parentJoinPath ? `${parentJoinPath}.${prop.name}` : `${meta.name}.${prop.name}`;
       qb.join(field, tableAlias, {}, 'leftJoin', path);
-      fields.push(...this.getFieldsForJoinedLoad(qb, meta2, relation.children, tableAlias, path));
+      const propFieldsAndOrderBy = this.getFieldsAndOrderByForJoinedLoad(qb, meta2, relation.children, tableAlias, path);
+      const propOrderBy = {};
+      if (prop.orderBy) {
+        for (const childField in prop.orderBy) {
+          propOrderBy[`${tableAlias}.${childField}`] = prop.orderBy[childField];
+        }
+      }
+      fields.push(...propFieldsAndOrderBy.fields);
+      orderBy = { ...orderBy, ...propOrderBy, ...propFieldsAndOrderBy.orderBy };
```

Comment:
you should not need to reassign the variable here. this is equal to the following, right?

```suggestion
      Object.assign(orderBy, { ...propOrderBy, ...propFieldsAndOrderBy.orderBy };
```